addons/featured-dangos-v2/userscript.js

```diff
@@ -0,0 +1,47 @@
+export default async function ({ addon, console, msg }) {
+  // Fetch projects from a studio
+  const fetchStudio = async (studio) => {
+    const response = await fetch(`https://api.scratch.mit.edu/studios/${studio}/projects/`);
+    return response.json();
+  };
+
+  const projects = await fetchStudio("33286407");
+
+  // Wait until Featured Projects row loads
+  const firstThumbnail = await addon.tab.waitForElement(".carousel .thumbnail");
+
+  // Get Featured Projects row <div>
+  const featuredProjectsRow = firstThumbnail.closest(".box");
+
+  // Clone Featured Projects row and its children
+  const featuredDangosRow = featuredProjectsRow.cloneNode(true);
+  featuredDangosRow.classList.add("sa-featured-dangos-row");
+
+  // At this point, the Featured Dangos row might be 5 "Project" placeholders linking to
+  // /projects/1, or possibly the actual featured projects. We will be handling both cases.
+
+  // Change row title
+  featuredDangosRow.querySelector(".box-header > h4").textContent = msg("row-title");
+
+  featuredDangosRow.querySelectorAll(".thumbnail").forEach((project, i) => {
+    // For each project
+    const { id, title, username } = projects[i];
+
+    // Link to correct project
+    project.querySelectorAll("a").forEach((link) => (link.href = `/projects/${id}/`));
+
+    // Use correct project thumbnail
+    project.querySelector("img").src = `//uploads.scratch.mit.edu/projects/thumbnails/${id}.png`;
+
+    // Use correct project title
+    project.querySelector("a[title]").textContent = title;
+    project.querySelector("a[title]").title = title;
+
+    // Use correct project author
+    project.querySelector(".thumbnail-creator").firstElementChild.innerText = username;
```

Comment:
`TypeError: Cannot read properties of null (reading 'firstElementChild')`

See comment above:
> At this point, the Featured Dangos row might be 5 "Project" placeholders linking to /projects/1, or possibly the actual featured projects. We will be handling both cases.

This is not handling the second case properly, crashing the addon.

Reply:
You can emulate this case by using Chrome Devtools' "network request blocking" feature to block `api.scratch.mit.edu/proxy/featured`

Reply:
Is it just to prevent the addon from crashing when Scratch's API isn't working like it should?

Reply:
No, it's a race condition that can depend on browser caching, Scratch server speed or the user's internet speed.
runAtComplete:true has very few guarantees in scratch-www pages.

Reply:
I recommended the "network request blocking" feature because there's no easy way to slow down a specific request.

Reply:
Ok, I'll need to use a different solution because for now I surround the username code with an if.

Reply:
That's good enough IMO, the original version of the addon didn't include the authors at all.

Reply:
This probably wouldn't happen in practice because the studio fetch is done first. My bad. But technically it was a bug.